plugins/pipelines/create.js

```diff
@@ -52,34 +52,40 @@ module.exports = () => ({
             // fetch the user
             return userFactory.get({ username })
                 // get the user permissions for the repo
-                .then(user => user.getPermissions(scmUrl))
-                // if the user isn't an admin, reject
-                .then(permissions => {
-                    if (!permissions.admin) {
-                        throw boom.unauthorized(`User ${username} is not an admin of this repo`);
-                    }
-                })
-                // see if there is already a pipeline
-                .then(() => pipelineFactory.get({ scmUrl }))
-                // if there is already a pipeline for the scmUrl, reject
-                .then(pipeline => {
-                    if (pipeline) {
-                        throw boom.conflict('scmUrl needs to be unique');
-                    }
-                })
-                // set up pipeline admins, and create a new pipeline
-                .then(() => {
-                    const admins = {};
+                .then(user => user.getPermissions(scmUrl)
+                    // if the user isn't an admin, reject
+                    .then(permissions => {
+                        if (!permissions.admin) {
+                            throw boom.unauthorized(
+                              `User ${username} is not an admin of this repo`);
+                        }
+                    })
+                    // see if there is already a pipeline
+                    .then(() => pipelineFactory.get({ scmUrl }))
```

Comment:
Since you changed this in data-schema, you'll prob. want to change this here as well.  Sadly this PR is getting bigger and bigger.  Sorry @catto  :(